### PR TITLE
msg/async: initialize worker in RDMAStack::create_worker() and drop Stack::num_workers 

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

**Review by yuyuyu101 — APPROVED** [review]

**Review by runsisi — APPROVED** [review]
lgtm

File: src/msg/async/Stack.cc [hunk 1]
```diff
@@ -121,17 +118,18 @@ void NetworkStack::start()
...
   }
 
-  for (unsigned i = 0; i < num_workers; ++i) {
-    if (workers[i]->is_init())
+  for (Worker* worker : workers) {
+    if (worker->is_init())
       continue;
-    std::function<void ()> thread = add_thread(i);
-    spawn_worker(i, std::move(thread));
+    std::function<void ()> thread = add_thread(worker);
```
Comment (runsisi) [inline src/msg/async/Stack.cc:124]: `thread` is a thread function, its naming is a little werid..

Comment (tchaikov) [reply src/msg/async/Stack.cc]: @runsisi eliminated the local variable. could you take another look?

File: src/msg/async/dpdk/DPDKStack.h [hunk 2]
```diff
@@ -254,12 +254,11 @@ class DPDKStack : public NetworkStack {
   }
 
  public:
-  explicit DPDKStack(CephContext *cct): NetworkStack(cct) {
-    funcs.resize(cct->_conf->ms_async_max_op_threads);
```
Comment (runsisi) [inline src/msg/async/dpdk/DPDKStack.h:258]: @tchaikov after this pr, `ms_async_max_op_threads` should be not used anymore, so maybe we could add a commit to get rid of it :)

```
$ grep -r ms_async_max_op_threads
doc/rados/configuration/ms-ref.rst:``ms_async_max_op_threads``
src/common/options.cc:    Option("ms_async_max_op_threads", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
src/common/legacy_config_opts.h:OPTION(ms_async_max_op_threads, OPT_U64)        // max number of worker processing threads for async messenger
src/msg/async/dpdk/DPDKStack.h:    funcs.resize(cct->_conf->ms_async_max_op_threads);
```

Comment (tchaikov) [reply src/msg/async/dpdk/DPDKStack.h:258]: @runsisi thanks for pointing this out. added and repushed.

**Pull request merged**